fetch-markdown.js

```diff
@@ -26,33 +26,31 @@ const REPORT_DIR = 'reports';
  * @returns - An object with the totals and the list of failures
  */
 export async function fetchMarkdown(project, site, indexUrl, cached = true) {
-    const totals = { success: 0, failure: 0 };
-    const failures = [];
+    const totals = { succeed: 0, failed: 0 };
+    const failed = [];
     const entries = await loadIndex(project, indexUrl, cached);
+    const mdFolder = `./${MD_DIR}/${project}`;
 
-    for (let i = 0; i < entries.length; i++) {
-        const entry = entries[i];
-        const markdown = await loadMarkdown(`${site}${entry}`, `./${MD_DIR}/${project}${entry}`, cached);
-
+    await loadMarkdowns(site, mdFolder, entries, cached, (markdown, entry, i) => {
```

Comment:
We might want to discuss how this is built at some point. 5 params, one of which is a callback, is a lot of params.

Reply:
Having more than 3 parameter can be sign that a function is doing too much but also can be unavoidable or even preferable. 
`loadMarkdowns` calls `loadMarkdown` so I need 3 parameters plus the callback. We could to reduce parameter count by pre-mapping the entries into a array of objects, `[{mdUrl, mdPath}, ...]`, but now we are passing a lot of redundant data with similar complexity.

We can move the callback to a named function, and that seems like a good idea.
Also we can have a separate functions for cached and uncached load markdowns, boolean parameter suggest this.
```js
loadMarkdowns(entries, site, callback)
loadMarkdownsCached(entries, mdFolder, site, callback)
```
Not sure how useful that is yet, we need to use this first.